src/YDialogSpy.cc

```diff
@@ -102,29 +125,77 @@ static void fillTree( YWidgetTreeItem * 		parent,
 
 
 
-struct YDialogSpyPrivate
+class YDialogSpyPrivate
 {
+public:
+
     YDialogSpyPrivate()
 	: targetDialog( 0 )
 	, spyDialog( 0 )
 	, widgetTree( 0 )
 	, propButton( 0 )
 	, propReplacePoint( 0 )
 	, propTable( 0 )
-	, closeButton( 0 )
 	{}
 
+    ~YDialogSpyPrivate();
+
     YDialog *		targetDialog;	// Dialog that is being inspected
     YDialog *		spyDialog;	// Debug dialog that shows widget data
     YTree *		widgetTree;	// Tree widget to show widget hierarchy
     YPushButton * 	propButton;
+    YMenuButton * 	addButton;
+    YPushButton * 	deleteButton;
+    YPushButton * 	upButton;
+    YPushButton * 	downButton;
     YReplacePoint *	propReplacePoint;
     YTable *		propTable;
-    YPushButton *	closeButton;
+    YMenuItem *exportMenu;
+
+    YWidget * selectedWidget();
+    void selectedWidgetChanged();
+    void refreshProperties();
+    bool toggleProperties();
+    void highlightWidget(bool enable = true);
+
+    void deleteWidget();
+    void addWidget(const std::string &type);
+    void editProperty();
+    void moveSelectedUp() { moveSelected(true); }
+    void moveSelectedDown() { moveSelected(false); }
+
+private:
+    void moveSelected(bool up);
```

Comment:
Using bool parameter is often criticized. Better would be a enum with values UP and DOWN.

On the other hand you have already moveSelectedUp and moveSelectedDown so it public API is understandable.